Makefile

```diff
@@ -0,0 +1,154 @@
+ROOT_DIR := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+
+export LLC := llc
+export CLANG := clang
+export OBJCOPY := llvm-objcopy
+
+CARGO := cargo
+RELEASE_VERSION ?= $(shell $(CARGO) metadata --no-deps --format-version=1 | jq -r '.packages | .[] | select(.name == "retis") | .version')
+RELEASE_NAME ?= $(shell $(CARGO) metadata --no-deps --format-version=1 | jq '.packages | .[] | select(.name=="retis") | .metadata.misc.release_name')
+
+PRINT = echo
+
+ifeq ($(V),)
+	Q=@
+	MAKE += -s
+	CARGO += -q
+define out_console
+	$(PRINT) -e "[$(1)]\t$(2)"
+endef
+
+.SILENT:
+endif
+
+ifeq ($(NOVENDOR),)
+    # This MUST be kept in sync with API_HEADERS under lib.rs in libbpf-sys
```

Comment:
That looks a bit dangerous. What about:

```
LIBBPF_API_HEADERS := $(shell sed -n "s/header!(\"\(.*\)\"),/\1/p" $(LIBBPF_SYS_LIBBPF_BASE_PATH)/src/lib.rs)
```

(untested)

Reply:
I was trying to avoid things like that because it's a bit fragile (if renamed or moved), but of course, feel free to disagree. 
The comment though is a bit misleading (it's not an all-or-nothing thing, but it's just what we need).
Here the only thing we should be careful of is that, if we have system headers and we miss exposing vendored ones, those would be "silently" used.

Reply:
> Being library headers (local to the actual libbpf) they should remain as they are, so it should be safe. The comment though is a bit misleading (it's not an all-or-nothing thing, but it's just what we need). 

Makes sense, and is less scary.

> Here the only thing we should be careful of is that, if we have system wide headers and we miss exposing vendored ones, those would be "silently" used.

But still a bit :)

> I was trying to avoid things like that because it's a bit fragile (if renamed or moved or gets fed in a way that makes it fail), but of course, feel free to disagree.

Yeah, I agree, both ways are not perfect. Maybe one solution would be to keep the list hardcoded and use the above shell command to check it matches `libbpf-sys` definition? This way we can detect changes.

Reply:
> > Being library headers (local to the actual libbpf) they should remain as they are, so it should be safe. The comment though is a bit misleading (it's not an all-or-nothing thing, but it's just what we need).
> 
> Makes sense, and is less scary.
> 
> > Here the only thing we should be careful of is that, if we have system wide headers and we miss exposing vendored ones, those would be "silently" used.
> 
> But still a bit :)
> 

yeah, I have no perfect solution if compiled on a system with libbpf

> > I was trying to avoid things like that because it's a bit fragile (if renamed or moved or gets fed in a way that makes it fail), but of course, feel free to disagree.
> 
> Yeah, I agree, both ways are not perfect. Maybe one solution would be to keep the list hardcoded and use the above shell command to check it matches `libbpf-sys` definition? This way we can detect changes.

IMHO, I'm not sure having both would make it better.
In such a case, we'd know there's a discrepancy, but that doesn't mean our programs will be compiled against older headers. Not sure if a warning is enough of a failure is required (and if required, maybe it's better to go back to parsing `header!()`).

I guess for this as well, given there will be a follow-up for detections, we could defer this there as well (https://github.com/retis-org/retis/pull/237#discussion_r1477049620)

Reply:
> > > I was trying to avoid things like that because it's a bit fragile (if renamed or moved or gets fed in a way that makes it fail), but of course, feel free to disagree.
> > 
> > 
> > Yeah, I agree, both ways are not perfect. Maybe one solution would be to keep the list hardcoded and use the above shell command to check it matches `libbpf-sys` definition? This way we can detect changes.
> 
> IMHO, I'm not sure having both would make it better. In such a case, we'd know there's a discrepancy, but that doesn't mean our programs will be compiled against older headers.

We can make the build fail. Yes the build might succeed and the output might be OK, but IMO in such case the build should be fixed so that it has the right expectations. Or am I missing a use case?

> I guess for this as well, given there will be a follow-up for detections, we could defer this there as well ([#237 (comment)](https://github.com/retis-org/retis/pull/237#discussion_r1477049620)).
> 
> WDYT?

Sure, we can do this by steps :)